pkg/engine/operation/graph/resource_node.go

```diff
@@ -252,6 +254,11 @@ func ParseImplicitRef(v reflect.Value, resourceIndex map[string]*models.Resource
 			}
 			v = tv
 		}
+		if secured {
```

Comment:
although `secured` can optimize some speed but not necessary, because IsSecured() has already checked whether this resource contains encrypted values. 

Delete `secured` can also reduce an input params of this func...

Reply:
No, you are wrong, `secured` comes from stack label, which means that value's format can be `ref+<secret store>`, but not secret ref, it's up to user, we cannot kidnap users

Reply:
Once `ref+<secret store>` is shown in user's code, that means this code is `secured`. Why should user add an extra config in stack label? Under what conditions users add `secured` label in stack and no secret ref in codes?

Reply:
yes, I learn this from other product, eg: pulumi following this rule. Besides, secured label is not created by my own, I saw it in previous encryption solution.